src/wallet/wallet.cpp

```diff
@@ -3022,6 +3011,30 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransac
     return true;
 }
 
+bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransactionRef& tx, CReserveKey& reservekey, CAmount& nFeeRet,
+                                int& nChangePosInOut, std::string& strFailReason, const CCoinControl& coin_control, bool sign)
+{
+    int nChangePosIn = nChangePosInOut;
+    CTransactionRef tx2 = tx;
+    bool res = CreateTransactionInternal(vecSend, tx, reservekey, nFeeRet, nChangePosInOut, strFailReason, coin_control, sign);
+    // try with avoidpartialspends unless it's enabled already
```

Comment:
I think I would prefer that we not do this -- I'm not sure that `-avoidpartialspends` works very well on wallets that have substantial address reuse, in particular I think you can devolve into cases where you'd produce giant transactions that take forever to sign and would never pass policy (or consensus) limits, in extreme cases.

I'm going to comment elsewhere about places I think we can mitigate this sort of pathological behavior, but as belt and suspenders I think if the user is not using this new feature, we should not fall back to trying it anyway.

Reply:
The code only selects the avoiding tx if its fee is lower than the original. I don't think you'd ever run into that case for someone with substantial address reuse.

Reply:
But even running the code on a wallet with a large number of outputs with the same scriptPubKey can be very slow.  Nothing stops the code from eg trying to generate a multi-megabyte transaction and then hit a huge slow down while trying to sign such a transaction, before the transaction gets rejected.  (I have an RPC test that demonstrates this possibility.)

Reply:
Now that there is a cap on the output group count, I think parts of this is mitigated, but I'm not sure I've convinced myself this is necessary. Perhaps a 3-state flag (try, on, off) with default=try, so people can disable it...

Reply:
Removing this. No harm in doing the try style later.